src/yacs-web/src/components/SelectedCourses.vue

```diff
@@ -0,0 +1,153 @@
+                    <button type="button" class="close text-muted" @click="removeCourse(course)">
+                        <span>&times;</span>
+                    </button>
+                    {{readableDate(course.date_start)}}-{{readableDate(course.date_end)}}
+
+                    <b-list-group flush>
+                        <b-list-group-item
+                            button
+                            v-for="section in course.sections"
+                            :key="section.crn"
+                            @click.stop="toggleCourseSection(course, section)"
+                            :style="{
+                                'border-left': section.selected ? `4px solid ${ColorService.getBorderColor(section)}` : 'none',
+                                'background-color': section.selected ? `${ColorService.getBackgroundColor(section)}` : 'white'
+                            }"
+                        >
+                            {{section.crn}} - {{section.sessions[0].section}}<br>
+
+                            <span
+                                v-for="courseSession in section.sessions"
+                                :key="courseSession.crn + courseSession.day_of_week + courseSession.time_start"
+                            >
+                                {{DAY_SHORTNAMES[courseSession.day_of_week + 1]}}: {{readableTime(courseSession.time_start)}} - {{readableTime(courseSession.time_end)}}
+                                <br>
+                            </span>
+
+                        </b-list-group-item>
+                    </b-list-group>
+                </b-card>
+            </b-card-group>
+        </b-col>
+    </b-row>
+</template>
+
+<script>
+import { 
+    TimeDateMixin 
+} from '@/mixins';
+
+import {
+    ColorService
+} from '@/services';
+
+import { faPaperPlane } from '@fortawesome/free-solid-svg-icons';
+
+export default {
+    name: 'SelectedCourses',
+    mixins: [
+        TimeDateMixin
+    ],
+    props: {
+        courses: Object,
+    },
+    data () {
+        return {
+            ColorService,
+            exportIcon: faPaperPlane,
+            ICS_DAY_SHORTNAMES: ["MO", "TU", "WE", "TH", "FR", "SA", "SU"],
+        }
+    },
+    methods: {
+        removeCourse (course) {
+            this.$emit('removeCourse', course);
+        },
+        toggleCourseSection (course, section) {
+            if (section.selected) {
+                this.$emit('removeCourseSection', section);
+            } else {
+                // Only allow selection of one section per course
+                this.$emit('removeAllCourseSections', course);
+                this.$emit('addCourseSection', course, section);
+            }
+        },
+        exportScheduleToIcs () {
+            let calendarBuilder = window.ics()
+            let semester;
+
+            for (const course of Object.values(this.courses)) {
+                for (const section of course.sections.filter(s => s.selected)) {
```

Comment:
As of now, this will give back only 1 course since you can only select 1 section as per the `toggleCourseSection`.

I like the idea of "future-proofing" code. But not sure if this would cause some confusion since it might lead one to believe you can select multiple sections of the same course.

(_Note:_ I believe in old yacs TAs wanted the ability to add multiple sections of the same course. Not sure what the verdict on that was.. )

Reply:
Good point, this will probably also change after the addition of 'possible schedules' based on selected course sections. But in the meantime I agree and will remove that constraint.